src/Security.php

```diff
@@ -27,15 +27,19 @@ protected static function heuristicScan($xml)
     /**
      * Scan XML string for potential XXE and XEE attacks
      *
-     * @param   string $xml
-     * @param   DomDocument $dom
-     * @param   int $libXmlConstants additional libxml constants to pass in
-     * @param   callable $callback the callback to use to create the dom element
-     * @throws  Exception\RuntimeException
+     * @param string            $xml
+     * @param DOMDocument|null  $dom
+     * @param int               $libXmlConstants additional libxml constants to pass in
+     * @param callable          $callback        the callback to use to create the dom element
+     *
      * @return  SimpleXMLElement|DomDocument|boolean
      */
-    private static function scanString($xml, DOMDocument $dom = null, $libXmlConstants, callable $callback)
-    {
+    private static function scanString(
+        string $xml,
+        DOMDocument|null $dom = null,
+        int $libXmlConstants,
+        callable $callback
+    ): SimpleXMLElement|DomDocument|bool {
```

Comment:
@gsteel Since there are breaking changes anyway, should we use that opportunity to change the parameter order as well? For example this method has the optional parameter as second instead of last item.

Reply:
laminas-xml is in security-only mode so it's unlikely that we'll be releasing a new major version. I'm not 100% sure on the rules around this so I'll have to check with the @laminas/technical-steering-committee …

Reply:
@SuperDJ 
> Since there are breaking changes anyway…

Please note, this package is in security-only maintenance mode. See:

- https://github.com/laminas/laminas-xml#laminas-xml
- https://getlaminas.org/packages-maintenance-status/

This means that there will be no further updates or changes. Adding version 8.4 of PHP is the only bigger change, aside from security updates.

Some more background informations: https://getlaminas.org/blog/2023-11-28-laminas-and-mezzio-supports-php-83.html

Reply:
@gsteel
You were faster... 😄